### ISSUE TITLE
Stop disturbing

### ISSUE THREAD

**Issue by diegosom** [hovancik/stretchly#189]
Hi, your app is awesome!

Sometimes a popup appear in the middle of writing a text. I don't want to lose the concentration, so I stop typing, switch to the mouse, close popup, and continue typing from the middle of the  word. It's so annoying and happens several every day.

Would be great if app would wait for completing the current work and shows the popup right after that. For example, waiting for a 5 second pause in typing could work.

**Comment by hovancik** [comment]
Hi, do you know that you can use shortcut to skip break? Based on your platform, it's `Cmd/Ctrl+X`

**Comment by diegosom** [comment]
@hovancik, you can disagree with me, but it also detracts from the working process.

**Comment by hovancik** [comment]
I do not disagree. I had people asking for this feature before and I though that it would be solved with notifications before breaks and with possibility to easy cancel break.

**Comment by diegosom** [comment]
@hovancik both of them interrupt user's work
Anyway it's just an idea, I'm not quiet sure if such keyboard/mouse tracking is possible at all and easy to implement

**Comment by hovancik** [comment]
I'm labeling this with RGSoC, so we can look at this with students during summer. I'm actually beginning to like the idea of waiting few seconds before starting a break.

**Comment by thankyou-xx** [comment]
What if there was a whitelist of the executables - if the whitelisted exe is in the active window, the notification will not appear(or, optionally, appear to the side of the screen). After the exe is closed, the notification could appear how many times you missed a break and need to make up for it. 
This is especially useful for high-immersion activities such as movie watching. I do realize one can set pause for +1 hours, but having to do so each time I want to watch something is a waste of time and attention.
Thanks for reading.

**Comment by tburny** [comment]
I would simply suggest a "typing with grace period" approach: Don't interrupt while the keyboard is active and wait at least X (10?) seconds for more keystrokes before interrupting. I know this is kind of very low-level, though.

**Comment by hovancik** [comment]
Thanks for feedback guys. 

Right now, I don't know about any way to track apps opened by user ;/ I was checking it because this was proposed before.  

What will probably be implemented is: after notification, when the time is for the break start, the icon will start flashing or something like that. App will wait for some time (let's say minute) until user is inactive (for let's say 10 seconds) and then start the break  (or after the minute of waiting). 

As it's bigger change, this will take some time. I am super busy but I hope students from RGSoC will be able to work on this or another feature.

**Comment by hovancik** [comment]
Hey, no news right now, we are wrapping up Rails Girls Summer of Code and after that I will need to think about next set of features to be developed.

**Comment by decadent** [comment]
BTW, this is exactly the problem that the proposal #144 solves: when you see a break warning with its own timeout, you can brace yourself to the oncoming break, instead of it popping up in your face. Somehow the notifications don't have the same effect for me—presumably because of the static pause between the notification and the break.

A microbreak is usually not too long to lay off the typing for those couple dozen seconds (sometimes even helping to clear the head a bit). For the full-on break, I like the method of Time Out and Workrave (iirc): I can postpone the break for another five minutes, to wrap up what I was doing.

I've coded that break warning for myself and am using Stretchly with it for >1.5 years now. I'll try to submit the code soon, though it's not quite production quality due to one bug.

IIRC I also implemented the detection of whether the user keeps typing during the warning, and was aborting the break in that case—but of course I promptly began abusing that feature, so turned it off.

----

Note also that simply postponing breaks while the user is typing wouldn't quite work for people or activities that are heavy on the keyboard use. E.g. personally, I'm using the keyboard in the browser and the file manager, moving through my notes with the keyboard, etc.—pretty much using it everywhere. And foreseeing a ‘solution’: I don't believe that an option would be better than a nudge for the user.

**Comment by hovancik** [comment]
There is ability to postpone break.TBH,  I am afraid to add more features around the start of break because the code is complicated as it is and if some bug happens the only way then might be to remove features...

**Comment by hovancik** [comment]
Also:  I don't believe in "I am in flow, I want to continue working". I think it's fallacy. It's more reasonable to take a break and let brain work in background. 

If people really think they are in flow and need to continue working, then they should change break/work periods in stretchly, and not try to work around default stretchly settings.

**Comment by hovancik** [comment]
Hi all, I am going through the old issues and as there is no plan right now to change how stretchly works, I am closing this issue. Thanks for your input!
